salt/modules/rabbitmq.py

```diff
@@ -111,15 +117,14 @@ def _get_rabbitmq_plugin():
     This works by taking the rabbitmq-server version and looking for where it
     seems to be hidden in /usr/lib.
     '''
-    rabbitmq = __context__.get('rabbitmq-plugins')
+    global RABBITMQ_PLUGINS
 
-    if rabbitmq is None:
+    if RABBITMQ_PLUGINS is None:
         version = __salt__['pkg.version']('rabbitmq-server').split('-')[0]
-        rabbitmq = ('/usr/lib/rabbitmq/lib/rabbitmq_server-{0}'
-                    '/sbin/rabbitmq-plugins').format(version)
-        __context__['rabbitmq-plugins'] = rabbitmq
+        RABBITMQ_PLUGINS = ('/usr/lib/rabbitmq/lib/rabbitmq_server-{0}'
+                            '/sbin/rabbitmq-plugins').format(version)
```

Comment:
I believe the point of `__virtual__` was to ensure that RabbitMQ is present on the system and succeed if it is and fail (return `False`) if it is not present. In figuring out if RabbitMQ is present or not, the path to rabbitmqctl is also calculated. So I think it makes sense to just put that in a global and then re-use the global where needed instead of added an extra function unnecessarily.

`RABBITMQ_PLUGINS` is where things get murky. On Windows, the path to this may also easily be calculated and put into this global variable. On certain Linux installations, it may (or may not) need extra logic. This extra logic is separated out in a separate function and is only used if a call that needs to configure the plugins is used (a subset of the supported functions). This logic requires a call into another module (via `__salt__['pkg.version']`) and so it may make sense to defer that call to be outside of `__virtual__`.